ast/src/parsed/mod.rs

```diff
@@ -442,7 +454,7 @@ impl<R> Expression<R> {
             Expression::Reference(_) | Expression::PublicReference(_) | Expression::String(_) => {
                 empty()
             }
-            Expression::Number(_, _) => empty(),
+            Expression::Number(_) => empty(),
```

Comment:
I'm wondering, should we implement `Children<Expression>` on all the structs and then uniformly call `.children()` on all of them here? Essentially saying that `Expression` only knows the collection of enums it has, but doesn't care what the inner structs look like and what they do. We could even encapsulate that in a macro "looping" over all the enum values.

Reply:
I just answered you something similar on MatchExpression's PR.  I agree with you, the implementation of children for expression should simply use the children of each variant.